server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -2,16 +2,69 @@ import express from 'express';
 import { GraphQLNonNull, GraphQLString } from 'graphql';
 
 import orderStatus from '../../../constants/order_status';
-import { canReject } from '../../../graphql/common/transactions';
+import { TransactionKind } from '../../../constants/transaction-kind';
+import { FEES_ON_TOP_TRANSACTION_PROPERTIES } from '../../../constants/transactions';
 import { purgeCacheForCollective } from '../../../lib/cache';
 import { notifyAdminsOfCollective } from '../../../lib/notifications';
 import models from '../../../models';
-import { Forbidden, NotFound, Unauthorized } from '../../errors';
+import { canReject } from '../../common/transactions';
+import { Forbidden, NotFound, Unauthorized, ValidationFailed } from '../../errors';
 import { refundTransaction as legacyRefundTransaction } from '../../v1/mutations/orders';
+import { AmountInput } from '../input/AmountInput';
 import { fetchTransactionWithReference, TransactionReferenceInput } from '../input/TransactionReferenceInput';
 import { Transaction } from '../interface/Transaction';
 
 const transactionMutations = {
+  addPlatformTipToTransaction: {
+    type: new GraphQLNonNull(Transaction),
+    description: 'Add platform tips to a transaction',
+    args: {
+      transaction: {
+        type: new GraphQLNonNull(TransactionReferenceInput),
+        description: 'Reference to the transaction in the platform tip',
+      },
+      amount: {
+        type: GraphQLNonNull(AmountInput),
+        description: 'Amount of the platform tip',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<typeof Transaction> {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to add a platform tip');
+      }
+
+      let transaction = await fetchTransactionWithReference(args.transaction, { throwIfMissing: true });
+
+      if (!req.remoteUser.isAdmin(transaction.HostCollectiveId)) {
+        throw new Unauthorized('Only host admins can add platform tips');
+      } else if (transaction.kind !== TransactionKind.ADDED_FUNDS) {
+        throw new ValidationFailed('Platform tips can only be added on added funds at the moment');
+      }
+
+      const isPlatformTipAvailable = await models.Transaction.findOne({
+        where: {
+          TransactionGroup: transaction.TransactionGroup,
+          kind: 'PLATFORM_TIP',
+        },
+      });
+
+      if (isPlatformTipAvailable) {
+        throw new Error('Platform tip is already set for this transaction group');
+      }
+
+      transaction = {
+        ...transaction,
+        data: { isFeesOnTop: true },
+        platformFeeInHostCurrency: args.amount.valueInCents,
```

Comment:
The goal of `AmountInput` is to offer a flexible API where people can use either float amounts or integers (cents). Using `getValueInCentsFromAmountInput` ensures that we support that flexible API, regardless of what parameters were given.

```suggestion
        platformFeeInHostCurrency: getValueInCentsFromAmountInput(args.amount),
```

Reply:
Thanks. 👍🏽